src/components/CheckBoxList/check-box-item.tsx

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { FunctionComponent } from 'react';
+import { Checkbox, ListItemIcon } from '@mui/material';
+import { SxProps } from '@mui/material';
+import OverflowableText from '../OverflowableText';
+
+export interface CheckBoxItemProps {
+    item: any;
+    checkBoxIconSx?: SxProps;
+    labelSx?: SxProps;
+    checked: boolean;
+    label: string;
+    onClick: (
+        event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+        item: any
+    ) => void;
+    secondaryAction?: React.ReactElement | undefined;
+    disabled: boolean | undefined;
+}
+
+const CheckBoxItem: FunctionComponent<CheckBoxItemProps> = ({
+    item,
+    checkBoxIconSx,
+    checked,
+    labelSx,
+    label,
+    onClick,
+    disabled = false,
+    secondaryAction,
+    ...props
+}) => {
+    return (
+        <>
+            <ListItemIcon sx={checkBoxIconSx}>
+                <Checkbox
+                    color={'primary'}
+                    edge="start"
+                    checked={checked}
+                    onClick={(event) => onClick(event, item)}
+                    disableRipple
+                    disabled={disabled}
+                    {...props}
+                />
+            </ListItemIcon>
+            <OverflowableText sx={labelSx} text={label} />
```

Comment:
Maybe we should use the classical code example from MUI demo here :
https://mui.com/material-ui/react-list/#checkbox 

```
<ListItem
  key={label} // maybe an id should be better here
  secondaryAction={secondaryAction}
>
  <ListItemButton role={undefined} onClick={(event) => onClick(event, item)} dense>
    <ListItemIcon>
      <Checkbox
        checked={checked.indexOf(value) !== -1}
        disableRipple
        inputProps={{ 'aria-labelledby': labelId }}
        {...props}
      />
    </ListItemIcon>
    <ListItemText id={label} >
       <OverflowableText sx={labelSx} text={label} />
    </ListItemText>
  </ListItemButton>
</ListItem>
```